pkg/loki/modules.go

```diff
@@ -1378,6 +1372,66 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 	}), nil
 }
 
+func (t *Loki) initV2QueryEngine() (services.Service, error) {
+	if !t.Cfg.Querier.EngineV2.Enable {
+		return nil, nil
+	}
+
+	store, err := t.createDataObjBucket("query-engine")
```

Comment:
Can we rename this function, the name confused me:

```suggestion
	store, err := t.createDataObjBucketClient("query-engine")
```

or 


```suggestion
	store, err := t.getDataObjBucket("query-engine")
```